esmvaltool/diag_scripts/thermodyn_diagtool/fourier_coefficients.py

```diff
@@ -0,0 +1,327 @@
+          retrieved. Time,level and lon dimensions
+          are retrieved;
+        - var_nc_fid: the id of the new NC dataset previously created;
+
+    PROGRAMMER(S)
+        Valerio Lembo (2018).
+    """
+    # Extract coordinates from NetCDF file
+    lons = nc_fid.variables['lon'][:]
+    var_nc_fid.createDimension('lon', len(lons))
+    var_nc_dim = var_nc_fid.createVariable('lon',
+                                           nc_fid.variables['lon'].dtype,
+                                           ('lon',))
+    for ncattr in nc_fid.variables['lon'].ncattrs():
+        var_nc_dim.setncattr(ncattr,
+                             nc_fid.variables['lon'].getncattr(ncattr))
+    var_nc_fid.variables['lon'][:] = lons
+
+
+def extr_plev(nc_fid, var_nc_fid):
+    """Extract plev coord. from NC files and save them to a new NC file.
+
+    Arguments:
+        - nc_f: the existing dataset, from where the metadata are
+          retrieved. Time,level and lon dimensions
+          are retrieved;
+        - var_nc_fid: the id of the new NC dataset previously created;
+
+    PROGRAMMER(S)
+        Valerio Lembo (2018).
+    """
+    plev = nc_fid.variables['plev'][:]
+    var_nc_fid.createDimension('plev', len(plev))
+    var_nc_dim = var_nc_fid.createVariable('plev',
+                                           nc_fid.variables['plev'].dtype,
+                                           ('plev',))
+    for ncattr in nc_fid.variables['plev'].ncattrs():
+        var_nc_dim.setncattr(ncattr,
+                             nc_fid.variables['plev'].getncattr(ncattr))
+    var_nc_fid.variables['plev'][:] = plev
+
+
+def extr_time(nc_fid, var_nc_fid):
+    """Extract time coord. from NC files and save them to a new NC file.
+
+    Arguments:
+        - nc_f: the existing dataset, from where the metadata are
+          retrieved. Time,level and lon dimensions
+          are retrieved;
+        - var_nc_fid: the id of the new NC dataset previously created;
+
+    PROGRAMMER(S)
+        Valerio Lembo (2018).
+    """
+    # Extract coordinates from NetCDF file
+    time = nc_fid.variables['time'][:]
+    # Using our previous dimension info, we can create the new dimensions.
+    var_nc_fid.createDimension('time', len(time))
+    var_nc_dim = var_nc_fid.createVariable('time',
+                                           nc_fid.variables['time'].dtype,
+                                           ('time',))
+    for ncattr in nc_fid.variables['time'].ncattrs():
+        var_nc_dim.setncattr(ncattr,
+                             nc_fid.variables['time'].getncattr(ncattr))
+    var_nc_fid.variables['time'][:] = time
+
+
+def varatts(w_nc_var, varname):
+    """Add attibutes to the variables, depending on their name.
+
+    Arguments:
+    - w_nc_var: a variable object;
+    - varname: the name of the variable, among ta, ua, va and wap.
+    """
+    if varname == 'ta':
+        w_nc_var.setncatts({'long_name': u"Air temperature",
```

Comment:
This kind of thing can be done much easier an more reliably by using the [`iris`](https://scitools.org.uk/iris/docs/latest/) library. It would be better if you could use that.